Rubberduck.Parsing/ComReflection/ComField.cs

```diff
@@ -29,12 +43,82 @@ public ComField(string name, VARDESC varDesc, int index, DeclarationType type)
+
+                if (ComVariant.TypeNames.TryGetValue(value.VariantType, out string typeName))
+                {
+                    _valueType = typeName;
+                }
+
+                if (value.VariantType.HasFlag(VarEnum.VT_ARRAY))
+                {
+                    IsArray = true;
+                }
             }
             else
             {
-                Debug.Assert(varDesc.varkind == VARKIND.VAR_PERINSTANCE);
-                ValueType = ComVariant.TypeNames.TryGetValue((VarEnum)varDesc.elemdescVar.tdesc.vt, out string typeName) ? typeName : "Object";
+                GetFieldType(varDesc.elemdescVar.tdesc, info);
+                if (!IsEnumMember || !ComProject.KnownEnumerations.TryGetValue(_enumGuid, out ComEnumeration enumType))
+                {
+                    return;
+                }
+                var member = enumType.Members.FirstOrDefault(m => m.Value == (int)DefaultValue);
+                _valueType = member != null ? member.Name : _valueType;
+            }
+        }
+
+        private static readonly HashSet<TYPEKIND> ReferenceTypeKinds = new HashSet<TYPEKIND>
+        {
+            TYPEKIND.TKIND_DISPATCH,
+            TYPEKIND.TKIND_COCLASS,
+            TYPEKIND.TKIND_INTERFACE
+        };
+
+        private void GetFieldType(TYPEDESC desc, ITypeInfo info)
+        {
+            var vt = (VarEnum)desc.vt;
+            TYPEDESC tdesc;
+
+            switch (vt)
+            {
+                case VarEnum.VT_PTR:
+                    tdesc = (TYPEDESC)Marshal.PtrToStructure(desc.lpValue, typeof(TYPEDESC));
+                    GetFieldType(tdesc, info);
+                    break;
+                case VarEnum.VT_USERDEFINED:
+                    int href;
+                    unchecked
+                    {
+                        href = (int)(desc.lpValue.ToInt64() & 0xFFFFFFFF);
+                    }
+                    try
+                    {
+                        info.GetRefTypeInfo(href, out ITypeInfo refTypeInfo);
+
+                        refTypeInfo.GetTypeAttr(out IntPtr attribPtr);
+                        var attribs = (TYPEATTR)Marshal.PtrToStructure(attribPtr, typeof(TYPEATTR));
+                        if (attribs.typekind == TYPEKIND.TKIND_ENUM)
+                        {
+                            _enumGuid = attribs.guid;
+                        }
+                        IsReferenceType = ReferenceTypeKinds.Contains(attribs.typekind);
+                        _valueType = new ComDocumentation(refTypeInfo, -1).Name;
+                        refTypeInfo.ReleaseTypeAttr(attribPtr);
+                    }
+                    catch (COMException) { }
+                    break;
+                case VarEnum.VT_SAFEARRAY:
+                case VarEnum.VT_CARRAY:
+                case VarEnum.VT_ARRAY:
+                    tdesc = (TYPEDESC)Marshal.PtrToStructure(desc.lpValue, typeof(TYPEDESC));
```

Comment:
Copy paste. :-)  These are all over the place, and I think it was because RD used to target 4.5. One of these days I'll find them all clean them up, but this at least keeps the usage fairly consistent.